### ISSUE TITLE
Query params not always correctly decoded

### ISSUE THREAD

**Issue by donchev7** [honojs/hono#1016]
Hey, I noticed when I am doing oAuth2 with google that sometimes google sends me query parameters such as:

```
{"state":"841c886d-14b7-4d82-9e03-dc5629cb473b","code":"4%2F0AVHEtk63juCeqrbpiPBBKuGcnk_QnmUzVLhZ5WFxkLsSeOy5IFGJAZwXb1345hYk4Kc4iw","scope":"email+profile+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.email+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.profile+openid","authuser":"1","prompt":"consent"}
```

or

```
{"state":"348c51b6-7cbc-40da-a48c-16c3b100659a","code":"4%2F0AVHEtk5mxLXlW660ngx3MWf4N1_tYox0lt2IC1mRNx398CJ3z4ab7gLP5kfWViyDNxv-hQ","scope":"email+profile+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.email+https%3A%2F%2Fwww.googleapis.com%2Fauth%2Fuserinfo.profile+openid","authuser":"1","prompt":"consent"}
```

in code when I do:

```
const { state, code } = c.req.query()
```

`code` looks like this: `4%2F0AVHEtk5mxLXlW660ngx3MWf4N1_tYox0lt2IC1mRNx398CJ3z4ab7gLP5kfWViyDNxv-hQ`

which is not correctly decoded. When I do:

```
decodeURIComponent(code)
```

I get the correct value: `4/0AVHEtk5mxLXlW660ngx3MWf4N1_tYox0lt2IC1mRNx398CJ3z4ab7gLP5kfWViyDNxv-hQ`

**Comment by donchev7** [comment]
Thanks @yusukebe  it works with v3.1.4 but I am waiting for [this](https://github.com/honojs/hono/issues/1022) before rolling it out to production.

**Comment by yusukebe** [comment]
@donchev7 

Released v3.1.5, including the fix.

**Comment by donchev7** [comment]
Great! Both cases work now, thank you.
